kernel/src/mm/pagetable.rs

```diff
@@ -977,7 +977,7 @@ struct RawPageTablePart {
 impl RawPageTablePart {
     /// Frees a level 1 page table.
     fn free_lvl1(page: &PTPage) {
-        for entry in page.entries {
+        for &entry in &page.entries {
```

Comment:
```suggestion
        for entry in page.entries.iter().copied() {
```
Nit: This could be a little more idiomatic.

Reply:
Doesn't this also cause the entry to be copied to the stack? I think omitting the `copied()` should do it

Reply:
Yes, that is better, thanks. Updated the PR.

Reply:
> Doesn't this also cause the entry to be copied to the stack? I think omitting the `copied()` should do it

Yes, it causes the entry to be copied to the stack, but it doesn't cause all of them to be copied to the stack at the same time like they were previously. A single entry is only 8 bytes, and so that isn't a problem, but previously, we copied the entire page table, which is 4096 bytes, and that is a problem.

We can't emit copying the entry entirely because `PTPage::from_entry` wants a copy, not a reference.

(Also, the single entry will likely be kept in a register instead of on the stack.)

Reply:
> Doesn't this also cause the entry to be copied to the stack? I think omitting the `copied()` should do it

Right, that reduces the stack usage even further (only 8 bytes, but still :) ). So updated again.